pkg/controller/common/settings/canonical_config.go

```diff
@@ -153,6 +153,15 @@ func (c *CanonicalConfig) HasKeys(keys []string) []string {
 	return has
 }
 
+// HasChildConfig returns true if c has a child config object below key.
+func (c *CanonicalConfig) HasChildConfig(key string) bool {
+	if c == nil {
+		return false
+	}
+	_, err := c.asUCfg().Child(key, -1, Options...)
+	return err == nil
```

Comment:
My thinking was that the two types of error that can be raised here are either ["type mismatch"](https://github.com/elastic/go-ucfg/blob/fc880abbe1f30b653d113da96a4a7e82743c0cc1/error.go#L72)
or ["missing field"](https://github.com/elastic/go-ucfg/blob/fc880abbe1f30b653d113da96a4a7e82743c0cc1/error.go#L62). We want both of those to be interpreted as "this config does not have a child config at this path"

I think the missing path case is self explanatory. The type mismatch would be if in my understand if we have a primitive value a the position the key is pointing to. 

```yaml
a: b
```

or your example
```yaml
x: {
```

I think in these cases I would not expect an error to be raised but just a `false` return value because `b` and `{` are not child configs but primitives. 

Let me know if I misunderstood your example or whether I am missing something (could totally be the case 😄 )

Reply:
Ok, got it 👍  You can ignore the suggestion 😄 , maybe just add a small comment to explain why we don't really care about the type of the returned error.